File: scala.console/src/main/java/org/netbeans/modules/scala/console/ScalaConsoleAction.java
package org.netbeans.modules.scala.console;

import java.awt.event.ActionEvent;
import javax.swing.AbstractAction;
import javax.swing.ImageIcon;
import org.openide.util.NbBundle;
import org.openide.util.Utilities;
import org.openide.windows.TopComponent;

/**
 * Action which shows Scala Console component.
 */
public class ScalaConsoleAction extends AbstractAction {
    
    public ScalaConsoleAction() {
        super(NbBundle.getMessage(ScalaConsoleAction.class, "CTL_ScalaConsoleAction"));
        putValue(SMALL_ICON,new ImageIcon(Utilities.loadImage(ScalaConsoleTopComponent.ICON_PATH, true)));
    }
    
    public void actionPerformed(ActionEvent evt) {
        TopComponent win = ScalaConsoleTopComponent.findInstance();
        win.open();
        win.requestActive();
    }
    
}

File: scala.console/src/main/scala/org/netbeans/modules/scala/console/old/ScalaConsoleAction.scala
package org.netbeans.modules.scala.console.old

import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.ImageIcon
import org.openide.util.ImageUtilities
import org.openide.util.NbBundle

/**
 * Action which shows Scala Console component.
 */
class ScalaConsoleAction extends AbstractAction(NbBundle.getMessage(classOf[ScalaConsoleAction], "CTL_ScalaConsoleAction")) {

  putValue(javax.swing.Action.SMALL_ICON, new ImageIcon(ImageUtilities.loadImage(ScalaConsoleTopComponent.ICON_PATH, true)))

  def actionPerformed(evt: ActionEvent) {
    val win = ScalaConsoleTopComponent.findInstance
    win.open
    win.requestActive
  }

}
